bedrock/cms/utils.py

```diff
@@ -0,0 +1,60 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import Http404
+
+from wagtail.models import Locale, Page
+
+from bedrock.base.i18n import split_path_and_normalize_language
+
+
+def get_page_for_request(*, request):
+    """For the given HTTPRequest (and its path) find the corresponding Wagtail
+    page, if one exists"""
+
+    lang_code, path, _ = split_path_and_normalize_language(request.path)
+    try:
+        locale = Locale.objects.get(language_code=lang_code)
+    except Locale.DoesNotExist:
+        locale = None
+
+    try:
+        page = Page.find_for_request(request=request, path=path)
+        if page and locale and locale != page.locale:
+            page = page.get_translation(locale=locale)
+
+    except (ObjectDoesNotExist, Http404):
+        page = None
+
+    return page
+
+
+def get_locales_for_cms_page(page):
+    # Patch in a list of CMS-available locales for pages that are
+    # translations, not just aliases
+
+    locales_available_via_cms = [page.locale.language_code]
+    try:
+        _actual_translations = (
+            page.get_translations()
+            .live()
+            .exclude(
+                id__in=[x.id for x in page.aliases.all()],
```

Comment:
I'd be curious if there's a SQL way to do this? Looks like this might hit the db for the list comprehension separately. Which is probably fine... sometimes successive small simple SQL queries can be better than one larger complex SQL query.

Reply:
Fair point about hitting the DB twice there. I'll refactor to use `Subquery` so we can at least keep it all ORM

Reply:
Oops, sorry, thought I approved with comments. I'll take a quick look over again and approve.